features/step_definitions/trema_steps.rb

```diff
@@ -0,0 +1,62 @@
+Given(/^I use OpenFlow 1\.0$/) do
+  @open_flow_version = :open_flow10
+end
+
+Given(/^I use OpenFlow 1\.3$/) do
+  @open_flow_version = :open_flow13
+end
+
+Given(/^a socket directory named "([^"]*)"$/) do |socket_directory|
+  step %(a directory named "#{socket_directory}")
+  ENV['TREMA_SOCKET_DIR'] = socket_directory
+end
+
+Then(/^a socket file named "([^"]*)" should exist$/) do |socket_file|
+  cd('.') do
+    expect(FileTest.socket?(socket_file)).to be_truthy
+  end
+end
+
+When(/^I trema run "([^"]*)"$/) do |controller_file|
+  controller_path = if controller_file.include?('/')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.